src/referrals.cpp

```diff
@@ -83,50 +110,54 @@ bool ReferralsViewCache::Exists(
         }
     }
 
-    if (auto ref = m_db->GetReferral(maybe_normalized, normalize_alias)) {
+    if (auto ref = m_db->GetReferral(maybe_normalized, false)) {
         LOCK(m_cs_cache);
-        referrals_index.insert(*ref);
-        alias_index[maybe_normalized] = ref->GetAddress();
+        InsertReferralIntoCache(*ref);
+
         return true;
     }
+
     return false;
 }
 
 void ReferralsViewCache::InsertReferralIntoCache(const Referral& ref) const
 {
     LOCK(m_cs_cache);
     referrals_index.insert(ref);
+
+    if (ref.alias.size()) {
```

Comment:
i explicitly didn't do this here because ref.alias is not normalized. beacons store aliases pre-normalized for display purposes if I recall. however, the index stores the keys normalized. I would revert back to what it was before.